src/app/_components/RecentBlocks/NewBlockPlaceholder.tsx

```diff
@@ -0,0 +1,77 @@
+import { Box, Icon, Stack } from '@chakra-ui/react';
+import { ArrowsClockwise } from '@phosphor-icons/react';
+import { useEffect, useRef, useState } from 'react';
+
+import { StacksApiSocketClient } from '@stacks/blockchain-api-client';
+
+import { useGlobalContext } from '../../../common/context/useGlobalContext';
+import { Text } from '../../../ui/Text';
+import { BTC_BLOCK_MIN_WIDTH, EMPTY_BTC_BLOCK_WIDTH } from './consts';
+
+export function NewBlockPlaceholder({
+  newestBtcBlockHeight,
+  refetch,
+}: {
+  newestBtcBlockHeight: number;
+  refetch: () => void;
+}) {
+  const [hasNewBlocks, setHasNewBlocks] = useState(false);
+  const { activeNetwork } = useGlobalContext();
+  const lastBlockHeightRef = useRef(newestBtcBlockHeight);
+
+  useEffect(() => {
+    lastBlockHeightRef.current = newestBtcBlockHeight;
+    setHasNewBlocks(false);
+  }, [newestBtcBlockHeight]);
+
+  useEffect(() => {
+    const socketClient = new StacksApiSocketClient({ url: activeNetwork.url });
```

Comment:
There is a stacks api socket client in global context. Did you consider using that?
  <GlobalContext.Provider
      value={{
        activeNetwork: networks[activeNetworkKey] || {},
        activeNetworkKey, // TODO: rename this to activeNetworkUrl. activeNetwork should be used instead of activeNetworkKey as having the information in both places is redundant
        addCustomNetwork,
        removeCustomNetwork,
        networks,
        apiClient: getApiClient(activeNetworkKey),

        stacksApiSocketClientInfo: {
          client: stacksApiSocketClient,
          connect: connectStacksApiSocket,
          disconnect: disconnectStacksApiSocket,
        },
      }}
    >
      {children}
    </GlobalContext.Provider>
  );